cmd/tesla-http-proxy/main.go

```diff
@@ -20,33 +21,45 @@ const (
 	defaultPort = 443
 )
 
-const warning = `
+const nonLocalhostWarning = `
 Do not listen on a network interface without adding client authentication. Unauthorized clients may
 be used to create excessive traffic from your IP address to Tesla's servers, which Tesla may respond
 to by rate limiting or blocking your connections.`
 
+type HttpProxyConfig struct {
+	keyFilename  string
+	certFilename string
+	verbose      bool
+	host         string
+	port         int
+	timeout      time.Duration
+}
+
+var (
+	httpConfig = &HttpProxyConfig{}
+)
+
+func init() {
+	flag.StringVar(&httpConfig.certFilename, "cert", "", "TLS certificate chain `file` with concatenated server, intermediate CA, and root CA certificates")
```

Comment:
Is the root CA certificate required? They usually aren't presented by the server.

Reply:
It should be included if the cert is signed by a CA. Running the HTTP proxy without a cert specified fails.

This PR isn't touching the help message but happy to tweak if this is not accurate.

```go
func http.ListenAndServeTLS(addr string, certFile string, keyFile string, handler http.Handler) error
// ListenAndServeTLS acts identically to ListenAndServe,
// except that it expects HTTPS connections. Additionally, files containing a
// certificate and matching private key for the server must be provided.
// If the certificate is signed by a certificate authority, the certFile should
// be the concatenation of the server's certificate, any intermediates, and the CA's certificate.
```